docs/LRM-feedback.md

```diff
@@ -0,0 +1,420 @@
+3. Meaning of Identifiers
+4. Expressions
+5. Statements 
+6. Statements
+7. Code Sample
+
+
+## 1: Introduction
+
+File Input Reinterpretation Engine (FIRE) is a scripting language inspired by AWK, bash, and other syntacically light languages. These languages are renowed for their ability to robustly extract, pattern-match, and manipulate text files. FIRE seeks to emulate this functionality with a more attractive, C-Family inspired syntax and intuitive semantics.
+
+FIRE is intended to be utilized with large sets of delimited data, like `csv` files. FIRE is also animated by the premise that files are first class citizens. 
+
+FIRE was built by a team of Columbia University undergraduates for Professor Stephen Edward's Programming Language and Translators course. FIRE is written in OCaml , utilizing libraries built in `C`, and leveraging the `LLVM` compiler back-end.
+
+## 2: Lexical Conventions 
+
+#### 2.1 Strings
+
+Strings are sequences of characters surrounded by double quotes `"`.
+As string has the type `str`.
+
+#### 2.2 Identifiers
+
+Identifiers are a sequence of characters consisting of uppercase or lowercase letters, digits, underscores 
+or dashes. 
+
+#### 2.3 Keywords
+
+The following identifiers are restricted from use:
+
+* `int`
+* `str`
+* `regx`
+* `if`
+* `else`
+* `elif`
+* `else`
+* `for`
+* `while`
+* `break`
+* `return`
+* `func`
+* `array`
+* `file`
+* `print`
+* `return`
+* `map`
+* `filter`
+* `true`
+* `false`
+* `null`
+* `void`
+
+
+#### 2.5 Regular Expressions
+
+Regular expressions are a special sequence of characters used for pattern matching surounded by single quotes `'` and preceded by the keyword `r'`. Regular expressions are of type `regx`.   
+
+#### 2.6 Comments
+
+FIRE only supports the use the block comments. Comments are initiated with the `/*` symbol and terminated by the `*/` symbol. Everything in between the symbols will be ignored by FIRE during compilation
+
+## 3: Meaning of Identifiers
+
+Identifiers are names that correlate to a single value, a function, or an array. The restrictions on valid identifiers are found in section 2.2.
+
+The scope of an identifier can be either global or local. A local identifier's scope is limited to inside the braces `{ ... }` in which they are declared, whereas global identifiers are declared outside any braces.
+
+## 4: Expressions
+
+#### 4.1 Primary Expressions
+
+Primary expressions in FIRE can either because accessing an array type or a function call. The primary expression is followed by an expression of either asignments in the case of a function call or an accessing of the datatypes in an array.
+
+1. `Primary Expression [ expression ]`
+2. `Primary Expression ( expression )`
+
+
+#### 4.2 Assignment Operator
+
+The assignment operators `=` returns the value of the expression that is evaluated on its right-hand side and stores it in the identifier on the left hand side. The scope of that identifier is described in [section 3](#Identifiers)
```

Comment:
How do you prevent against malformed expressions such as ```x + 3 = x```

You can fix this by either making an assignment a statement (most straightforward), or by breaking down expressions into r-values and l-values (how C does it, but this can get a little complicated).